olp-cpp-sdk-core/src/http/curl/NetworkCurl.cpp

```diff
@@ -284,15 +284,13 @@ void NetworkCurl::Deinitialize() {
   }
 
   std::lock_guard<std::mutex> init_lock(init_mutex_);
+  // We are trying to stop the very thread we are in. This is not recommended,
```

Comment:
You cannot join() from the same thread. And based on the `if (thread_.get_id() != std::this_thread::get_id())` guard you also do not stop from the same thread you are in.
The comment should be changed to reflect that we should not stop from the same thread we are in.